### PR TITLE
add possibility to insert svg files from external source to Icon

### PR DESCRIPTION
**Pull request by ElaLukaszewicz**
image
You can insert external svg file through 'customSvg' property in Icon

### CODE REVIEW

File: src/components/icons/Icon.jsx [hunk 1]
```diff
@@ -68,24 +68,30 @@ export const ICON_COLOR = {
...
   }, className);
   const iconType = `#icon-${type}`;
 
   return (
-    <svg {...props} className={iconClass}>
-      <use xlinkHref={iconType} />
-    </svg>
+    <React.Fragment>
+      {customSvg ?
+        <div {...props} className={iconClass} dangerouslySetInnerHTML={{__html: customSvg}} /> :
```
Comment (tomasz-sodzawiczny) [inline src/components/icons/Icon.jsx]: Why `dangerouslySetInnerHTML`?
Can we just accept the `<svg />` react element?

File: src/components/icons/Icon.spec.jsx [hunk 2]
```diff
@@ -2,23 +2,29 @@ import React from 'react';
...
     <Icon type={TYPE.ANSWER} />
   );
 
-  expect(icon.hasClass('sg-icon')).toEqual(true);
+  expect(icon.find('svg')).toHaveLength(1);
+  expect(icon.find('svg').hasClass('sg-icon')).toEqual(true);
   expect(icon.find('use')).toHaveLength(1);
 });
 
-test('error when no type', () => {
```
Comment (vergilius) [inline src/components/icons/Icon.spec.jsx]: hm, we should still get an error if no type or custom svg is passed

File: src/components/icons/pages/icons.jsx [hunk 3]
```diff
@@ -18,6 +18,21 @@ const icons = () => (
       </ContrastBox>
     </DocsBlock>
 
+    <DocsBlock info="Custom SVG">
+      <ul className="icons-list">
+        <li className="icons-list__element">
+          <Icon
+            type={TYPE.FRIENDS}
+            color={ICON_COLOR.PEACH}
+            customSvg='<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
```
Comment (vergilius) [inline src/components/icons/pages/icons.jsx]: remove whitespaces?

File: src/components/icons/Icon.jsx [hunk 4]
```diff
@@ -68,24 +68,30 @@ export const ICON_COLOR = {
 
 export const SIZE = [120, 118, 96, 94, 64, 62, 48, 46, 38, 32, 30, 26, 24, 22, 20, 18, 16, 14, 10, 8];
 
-const Icon = ({color, size = 24, type, className, ...props}) => {
+const Icon = ({color, size = 24, type, customSvg, className, ...props}) => {
```
Comment (tomasz-sodzawiczny) [inline src/components/icons/Icon.jsx]: It seems like we could use `children` for that:
```jsx
<Icon>
  <svg>
    <use ...>
  </svg>
</Icon>
```

File: src/components/icons/Icon.jsx [hunk 5]
```diff
@@ -68,24 +68,52 @@ export const ICON_COLOR = {
...
     [`sg-icon--x${size}`]: size
   }, className);
   const iconType = `#icon-${type}`;
 
   return (
-    <svg {...props} className={iconClass}>
-      <use xlinkHref={iconType} />
-    </svg>
+    <div {...props} className={iconClass}>
+      {type ?
```
Comment (tomasz-sodzawiczny) [inline src/components/icons/Icon.jsx]: Maybe an error would be nice, if both are provided - someone might spend hours debugging in the future

File: src/components/icons/_icons.scss [hunk 6]
```diff
@@ -46,5 +46,10 @@ $iconSizes: 120, 118, 96, 94, 64, 62, 48, 46, 38, 32, 30, 26, 24, 22, 20, 18, 16
       }
     }
 
+    &__svg {
+      width: 100%;
+      height: 100%;
+      display: block;
```
Comment (tomasz-sodzawiczny) [inline src/components/icons/_icons.scss]: Is that necessary?

**Pull request merged**